SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityMesosStatusUpdateHandler.java

```diff
@@ -344,6 +346,35 @@ private StatusUpdateResult unsafeProcessStatusUpdate(
           taskIdObj
         );
         return StatusUpdateResult.KILL_TASK;
+      } else if (
+        maybeTaskHistory.isPresent() &&
+        status.getReason() == Reason.REASON_AGENT_REREGISTERED
+      ) {
+        boolean lbRemovalStarted = maybeTaskHistory
+          .get()
+          .getLoadBalancerUpdates()
+          .stream()
+          .anyMatch(
+            lbUpdate ->
+              lbUpdate.getMethod() == LoadBalancerMethod.CANCEL ||
+              lbUpdate.getMethod() == LoadBalancerMethod.DELETE
+          );
+        if (lbRemovalStarted) {
+          taskManager.createTaskCleanup(
+            new SingularityTaskCleanup(
+              null,
+              TaskCleanupType.PRIORITY_KILL,
+              System.currentTimeMillis(),
+              taskIdObj,
+              Optional.of(
+                "Agent re-registered after load balancer removal started. Task cannot be reactivated."
+              ),
+              Optional.empty(),
+              Optional.empty()
+            )
+          );
+          return StatusUpdateResult.IGNORED;
```

Comment:
Also not sure of this, but since we want to clean gracefully I figure we don't want mesos to touch it.